src/shared/components/community/sidebar.tsx

```diff
@@ -166,9 +168,14 @@ export class Sidebar extends Component<SidebarProps, SidebarState> {
   sidebar() {
     const {
       community: { name, ap_id, id, posting_restricted_to_mods, visibility },
-      community_actions: { received_ban_at } = {},
+      community_actions: { received_ban_at, follow_state } = {},
     } = this.props.community_view;
 
+    const visibility_label = community_visibility_label(visibility);
+    const visibility_description = (visibility_label +
+      "_desc") as NoOptionI18nKeys;
+    const can_view_community =
+      visibility !== "Private" || follow_state === "Accepted";
```

Comment:
I think you used this logic elsewhere, so it might be a good idea to add it as a utils helper function.